autorest/azure/async.go

```diff
@@ -348,6 +348,10 @@ func (pt *pollingTrackerBase) initializeState() error {
 	case http.StatusOK:
 		if ps := pt.getProvisioningState(); ps != nil {
 			pt.State = *ps
+			if pt.hasFailed() {
+				pt.updateErrorFromResponse()
+				return pt.pollingError()
```

Comment:
was wondering is it possible to just have this part of pollingError() call ? i.e. are there situations where we odnt want to call `updateErrorFromResponse` when we call pollingError ? and if so is that discernable based on `pollingTrackerBase` state ?

Reply:
It probably could be refactored (it's likely there was more differing state during the initial implementation) but given the finicky nature of the LRO state machine I'm a little hesitant to change it since things are working as is.